event-sourcing/event-repository/event-repository-jdbc/src/test/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventstream/EventStreamJdbcRepositoryIT.java

```diff
@@ -0,0 +1,260 @@
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final boolean exists = jdbcRepository.recordExists(10L);
+        assertFalse(exists);
+    }
+
+    @Test
+    public void shouldReturnHeadRecords() throws InvalidSequenceIdException, SQLException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+
+        final List<EventStream> streamOfStreams = jdbcRepository.head(2).collect(toList());
+        assertThat(streamOfStreams, hasSize(2));
+
+        assertThat(streamOfStreams.get(0).getStreamId(), is(streamId7));
+        assertThat(streamOfStreams.get(0).getSequenceNumber(), is(7L));
+
+        assertThat(streamOfStreams.get(1).getStreamId(), is(streamId6));
+        assertThat(streamOfStreams.get(1).getSequenceNumber(), is(6L));
+    }
+
+    @Test
+    public void shouldReturnNextRecords() throws InvalidSequenceIdException, SQLException {
+        final UUID streamId1 = randomUUID();
+        final UUID streamId2 = randomUUID();
+        final UUID streamId3 = randomUUID();
+        final UUID streamId4 = randomUUID();
+        final UUID streamId5 = randomUUID();
+        final UUID streamId6 = randomUUID();
+        final UUID streamId7 = randomUUID();
+        final UUID streamId8 = randomUUID();
+
+        jdbcRepository.insert(eventStreamOf(streamId1, 1L));
+        jdbcRepository.insert(eventStreamOf(streamId2, 2L));
+        jdbcRepository.insert(eventStreamOf(streamId3, 3L));
+        jdbcRepository.insert(eventStreamOf(streamId4, 4L));
+        jdbcRepository.insert(eventStreamOf(streamId5, 5L));
+        jdbcRepository.insert(eventStreamOf(streamId6, 6L));
+        jdbcRepository.insert(eventStreamOf(streamId7, 7L));
+
+        final List<EventStream> streamOfStreams = jdbcRepository.next(3L, 2).collect(toList());
```

Comment:
maybe we hard code 2 as a constant PAGE_SIZE. Then it is clearer...

Reply:
Done